application/marketing/marketing_test.go

```diff
@@ -0,0 +1,165 @@
+package marketing
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/silinternational/wecarry-api/domain"
+	"github.com/silinternational/wecarry-api/models"
+)
+
+// TestSuite establishes a test suite for domain tests
+type TestSuite struct {
+	suite.Suite
+}
+
+// Test_TestSuite runs the test suite
+func Test_TestSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (ts *TestSuite) Test_callApi() {
+	t := ts.T()
+
+	mux := http.NewServeMux()
+	srv := httptest.NewServer(mux)
+	defer srv.Close()
+
+	mux.HandleFunc("/", apiRequestHandler)
+
+	type args struct {
+		apiRequests []ApiRequest
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "callApi tests",
+			wantErr: false,
```

Comment:
Did you want to add one or more tests where `wantErr` is true?

Reply:
sure, I added one for that